src/commands/download_file.ts

```diff
@@ -0,0 +1,74 @@
+import { EID } from 'ardrive-core-js';
+// import { statSync } from 'fs';
+// import { join as joinPath, parse as parseFile, resolve as resolvePath } from 'path';
+// import { cliArDriveAnonymousFactory, cliArDriveFactory } from '..';
+import { CLICommand, ParametersHelper } from '../CLICommand';
+import { CLIAction } from '../CLICommand/action';
+import {
+	DriveIdParameter,
+	DrivePrivacyParameters,
+	DryRunParameter,
+	FileIdParameter,
+	LocalFilePathParameter
+} from '../parameter_declarations';
+
+// const fileCountRegExp = /_([0-9]+)$/;
+
+// function getFullFilePath(path: string, defaultFileName: string): string {
+// 	const fullLocalFilePath = resolvePath(path);
+// 	const parsedFile = parseFile(fullLocalFilePath);
+// 	try {
+// 		const fileStat = statSync(fullLocalFilePath);
+// 		if (fileStat.isDirectory()) {
+// 			return joinPath(fullLocalFilePath, defaultFileName);
+// 		} else {
+// 			// const fileName = parsedFile.name;
+// 			// const fileNameMatch = fileName.match(fileCountRegExp);
+// 			// const fileCount = fileNameMatch ? fileNameMatch[0] : 0;
+// 			// const fileExtension = parsedFile.ext;
+// 			console.log(`TODO: conflict name resolution for local files`);
+// 			console.log(`Overriding "${fullLocalFilePath}"`);
+// 			return `${fullLocalFilePath}`;
+// 		}
+// 	} catch (e) {
+// 		const parentLocalFolder = parsedFile.dir;
+// 		const dirStat = statSync(parentLocalFolder);
+// 		if (dirStat.isDirectory()) {
+// 			// The file does not exist, but the parent folder IS a directory
+// 			return fullLocalFilePath;
+// 		}
+// 		throw new Error(`No such file or directory: ${parentLocalFolder}`);
+// 	}
+// }
+
+new CLICommand({
+	name: 'download-file',
+	parameters: [FileIdParameter, LocalFilePathParameter, DryRunParameter, DriveIdParameter, ...DrivePrivacyParameters],
```

Comment:
Is `driveId`  meant to be required for downloading a file?

Also, perhaps localFilePath is optional. I would expect it to use `./` by default

Reply:
`driveId` is not required, but we use it in the `getDriveKey` method

Reply:
Now we default `localFilePath` to `./`: e5b74a090089b0af89eb7b138ac9589d59c68a9a